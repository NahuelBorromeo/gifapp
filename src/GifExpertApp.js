import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
// import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = ['Coder'] }) => {
    // No se hace con const a menos que el array no vaya a cambiar
    // const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];

    const [categories, setCategories] = useState( defaultCategories );

    // const handleAdd = () => {
    //     const newElement = prompt('QuÃ© elemento quiere agregar?');
    //     setCategories([...categories, newElement]);
    //     //Otra forma
    //     //setCategories( cats => [...cats, newElement]);
    // }

    return (
        <>
          <h2>GifExpertApp</h2>
          <AddCategory 
              setCategories={ setCategories }
          /> 
          <hr />  

          <ol>
              {
                  categories.map( (category) => 
                        <GifGrid 
                            key= { category }
                            category={ category }
                        />
                      )
              } 
          </ol>

        </>                                                                                                                                           
    )
}

export default GifExpertApp
