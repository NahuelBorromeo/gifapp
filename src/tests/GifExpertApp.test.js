import GifExpertApp from "../GifExpertApp";
import React from 'react';
import { shallow } from 'enzyme';

describe('Pruebas en <GifExpertApp />', () => {
    
    test('should to do match with the snapshot', () => {
        
        const wrapper = shallow(<GifExpertApp />);

        expect( wrapper ).toMatchSnapshot();

    });

    test('should to show a categories list', () => {
        
        const categories = ['One punch', 'Dragon ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    })
    
    


})
