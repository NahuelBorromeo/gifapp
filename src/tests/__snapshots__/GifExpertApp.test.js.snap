// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> should to do match with the snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> should to show a categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;
